### ISSUE TITLE
如何蒸馏不分层的新特征？

### ISSUE THREAD

**Issue by catqaq** [airaria/TextBrewer#93]
工具非常棒！
@airaria [您前面说明了如何自定义特征，并添加到中间层匹配里面去。](https://github.com/airaria/TextBrewer/issues/21)
我们使用过程中遇到一个小问题，如果新特征是不分层的，就像logits那样，该如何蒸馏该特征呢？使用CustomMatch吗，但是我们暂时没有发现比较简明的示例。目前我们的做法是使用新特征来作为logits（在满足size要求的情况下），相当于复用了你们预设的logits字段，但是使用的是我们自己定义的新特征。请问有更好的处理方式吗？
具体的，比如模型输出了三个特征：se(sentence_embedding), hidden, attention，现在想蒸馏se这个特征，目前的做法是占用logits，令'logits': model_outputs["se"]，倒是能用，但是感觉有些奇怪。

**Comment by airaria** [comment]
可以试着用intermediate_matches模拟你们的自定义不分层特征：

先为自定义特征随便起个名字，比如CustomFeature,
intermediate_matches这么写，把自定义特征伪装成第0层:
```python
intermediate_matches = [
{"layer_T":0, "layer_S":0, "feature":"CustomFeature", "loss":"CustomLoss", "weight": 1},
....]
```

修改adaptor使得返回的字典包含“CustomFeature”字段即可，对应的value用列表包起来，这样上面的``"layer_T":0``就正好可以取到其中的特征：
```python
def adatpor(batch, model_outputs):
    ...
    return {'CustomFeature': [CustomFeature], ... }
```
如果已有的loss不能满足你们需求，那需要仿照textbrewer.losses中loss的定义，自定义loss，然后添加到MATCH_LOSS_MAP即可：
```python
TextBrewer.MATCH_LOSS_MAP['CustomLoss'] = YourCustomLoss
```
参考/src/textbrewer/presets.py文件结尾的注释

**Comment by catqaq** [comment]
感谢回复，可用！
昨天看了下内部取特征的逻辑和post_adaptor，也看到了[logits]这种操作，将对应的特征用list包起来，与您介绍的方式是一致的。确实可以实现我们需要的功能。
